profiling/space-time-stack/kp_space_time_stack.cpp

```diff
@@ -66,7 +72,7 @@ Space get_space(SpaceHandle const& handle) {
   return SPACE_HOST;
 }
 
-const char* get_space_name(int space) {
+const char* get_space_name(const Space space) {
```

Comment:
I looked over the implementation and can't help asking what is the benefit of the switch case over a simple
```C++
char *space_names[] = {
  "HOST",
  "CUDA",
  ...
};
```
especially now that you pin their value in the enumeration declaration?
(not asking you to change here)